### PR TITLE
kie-issues#2549: Decision Service cannot be recognized in columns of decision table

### LINKED ISSUE #2549: Decision Service cannot be recognized in columns of decision table
It relates to https://github.com/apache/incubator-kie-drools/issues/6058

image
Decision Service cannot be identified inside the input/output column of decision table. 

DMN Validator returns `DECISION_TABLE_ANALYSIS_ERROR: Skipped Decision Table Analysis of table 'PayersByCountry' because: Name of function is not instance of NameRefNode!FunctionInvocationNode{CountryAD.CountryAD(Input)}
`

The same decision as `FEEL` or `Context` can be recognised DS functions without any issues.
image
image

### PR DESCRIPTION
**Pull request by danielzhe**
Closes: https://github.com/apache/incubator-kie-tools/issues/2549

It closes the ticket related to the colorize and autocomplete only.

https://github.com/user-attachments/assets/0059f31e-8eee-4426-8c3a-1b58ccc0a4df

### CODE REVIEW

File: packages/feel-input-component/tests/semanticTokensProvider.test.ts [hunk 1]
```diff
@@ -22,6 +22,10 @@ import { BuiltInTypes, DmnDefinitions, FeelVariables } from "@kie-tools/dmn-feel
 
 import * as Monaco from "@kie-tools-core/monaco-editor";
 import { Element } from "@kie-tools/feel-input-component/dist/themes/Element";
+import * as fs from "fs";
+import * as path from "path";
+import { getMarshaller } from "@kie-tools/dmn-marshaller";
+//import {getMarshaller} from "@kie-tools/dmn-marshaller/d"
```
Comment (jomarko) [inline packages/feel-input-component/tests/semanticTokensProvider.test.ts]: leftover probably?

File: packages/feel-input-component/tests/semanticTokensProvider.test.ts [hunk 2]
```diff
@@ -286,6 +290,149 @@ ThatShouldFailWhenBreakLine`,
...
+        ...getMonacoSemanticToken({
+          startLineRelativeToPreviousLine: 0,
+          startIndexRelativeToPreviousStartIndex: "LocalInput".length + ") + ".length,
+          tokenLength: "MyIncludedModel.RemoteInput".length,
+        }),
+      ];
+
+      for (let i = 0; i < expected.length; i++) {
+        expect(semanticMonacoTokens?.data[i]).toEqual(expected[i]);
+      }
```
Comment (jomarko) [inline packages/feel-input-component/tests/semanticTokensProvider.test.ts:376]: can't we simplify?:
```
expect(semanticMonacoTokens?.data).toEqual(expected)
```

see: https://jestjs.io/docs/expect#toequalvalue

**Review by jomarko — COMMENTED** [review]
Thank you @danielzhe , during my manual check, I didn't find an issue. I put some comments to code inline. Please let me know what do you think about them.

File: packages/feel-input-component/tests/semanticTokensProvider.test.ts [hunk 3]
```diff
@@ -377,3 +524,8 @@ function createModelMockForExpression(expression: string) {
     getLinesContent: jest.fn().mockReturnValue(expression.split("\n")),
   };
 }
+
+function getDmnModelFromFilePath(filePath: string) {
```
Comment (tiagobento) [inline packages/feel-input-component/tests/semanticTokensProvider.test.ts]: https://github.com/apache/incubator-kie-issues/issues/783#issuecomment-1871525975

File: packages/feel-input-component/tests/semanticTokensProvider.test.ts (hunk 2 above)
Comment (danielzhe) [reply packages/feel-input-component/tests/semanticTokensProvider.test.ts:376]: No, their types are different and it fails if we do this way:
```
Error: expect(received).toEqual(expected) // deep equality

Expected: [0, 0, 20, 6, 0, 0, 21, 10, 5, 0, …]
Received: serializes to the same string
<Click to see difference>
```

**Comment by danielzhe** [comment]
@jomarko @tiagobento Changes applied! Ready for you.

**Review by tiagobento — APPROVED** [review]

**Review by jomarko — APPROVED** [review]
Thank you @danielzhe

**Pull request merged**